DataSetExplorer/Core/DataSets/CaDETToDataSetProjectBuilder.cs

```diff
@@ -151,44 +150,67 @@ private static List<CoupledClassStrength> CountCouplingStrength(List<CaDETClass>
+                _classCouplings[coupledClassStrength.CoupledClass].Add(new CoupledClassStrength(instance, coupledClassStrength.CouplingStrength, coupledClassStrength.CouplingType));
+                return; 
             }
 
             var references = new List<CoupledClassStrength>();
-            references.Add(new CoupledClassStrength(instance, coupledClassStrength.CouplingStrength));
+            references.Add(new CoupledClassStrength(instance, coupledClassStrength.CouplingStrength, coupledClassStrength.CouplingType));
             _classCouplings.Add(coupledClassStrength.CoupledClass, references);
         }
 
         private List<RelatedInstance> FindRelatedInstances(CaDETClass c)
         {
             var relatedInstances = new List<RelatedInstance>();
-            if (c.Parent != null) relatedInstances.Add(new RelatedInstance(c.Parent.FullName, GetCodeUrl(c.Parent.FullName), RelationType.Parent, 1));
+            if (c.Parent != null)
+            {
+                var couplingTypeAndSt = new Dictionary<CouplingType, int>();
+                couplingTypeAndSt.Add(CouplingType.Parent, 1);
+                relatedInstances.Add(new RelatedInstance(c.Parent.FullName, GetCodeUrl(c.Parent.FullName), RelationType.Parent, couplingTypeAndSt));
+            }
             relatedInstances.AddRange(FindReferencedInstances(c));
             relatedInstances.AddRange(FindInstancesThatReference(c));
             return relatedInstances;
         }
 
         private List<RelatedInstance> FindReferencedInstances(CaDETClass referencingClass)
         {
-            return GetReferencedInstances(referencingClass).Select(cc => new RelatedInstance(cc.CoupledClass.FullName,
-                GetCodeUrl(cc.CoupledClass.FullName), RelationType.Referenced, cc.CouplingStrength)).ToList();
+            var relatedInstances = new List<RelatedInstance>();
+            GetReferencedInstances(referencingClass).ForEach(cc => CoupledClassToRelatedInstance(relatedInstances, cc, RelationType.Referenced));
+            return relatedInstances;
+        }
+
+        private void CoupledClassToRelatedInstance(List<RelatedInstance> relatedInstances, CoupledClassStrength cc, RelationType relationType)
+        {
+            var index = relatedInstances.FindIndex(i => i.CodeSnippetId.Equals(cc.CoupledClass.FullName));
+            if (index != -1)
+            {
+                relatedInstances[index].CouplingTypeAndStrength.Add(cc.CouplingType, cc.CouplingStrength);
+            }
+            else
+            {
+                var couplingTypeAndSt = new Dictionary<CouplingType, int>();
+                couplingTypeAndSt.Add(cc.CouplingType, cc.CouplingStrength);
+                relatedInstances.Add(new RelatedInstance(cc.CoupledClass.FullName,
+                    GetCodeUrl(cc.CoupledClass.FullName), relationType, couplingTypeAndSt));
+            }
         }
 
         private List<RelatedInstance> FindInstancesThatReference(CaDETClass referencedClass)
```

Comment:
This class is slowly growing into a god. Should these coupling items be part of the code model? You don't need to resolve this now, but maybe create an issue.

Reply:
https://github.com/Clean-CaDET/platform/issues/99